lxd-migrate/main_netcat.go

```diff
@@ -51,22 +50,26 @@ func (c *cmdNetcat) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	// We'll wait until we're done reading from the socket
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-
+	errs := make(chan error, 3)
 	go func() {
-		io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
-		conn.Close()
-		wg.Done()
+		_, err := io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
+		errs <- err
+
+		errs <- conn.Close()
+
+		defer close(errs)
 	}()
 
 	go func() {
-		io.Copy(eagain.Writer{Writer: conn}, eagain.Reader{Reader: os.Stdin})
+		_, err := io.Copy(eagain.Writer{Writer: conn}, eagain.Reader{Reader: os.Stdin})
+		errs <- err
```

Comment:
I think you still need to use the wait group to coodinate when these two go routines are both done before closing the `errs` channel as its quite possible in the current form you'll panic LXD because if the first go routine stops first and closes the `errs` channel, and then the 2nd go routine will finish io.Copy and writes it return value to the closed `errs` channel then LXD is going to panic.

The other option is to not run the 2nd io.Copy in its own go routine and instead use its blocking property to hold up the main thread until the it returns due to error or because the connections have been closed.

E.g. similar to https://github.com/lxc/lxd/blob/master/lxc/file.go#L1059-L1075

Reply:
I've blocked the main thread on io.Copy as you suggested.